pkg/controller/plan/kubevirt.go

```diff
@@ -893,6 +893,35 @@ func (r *KubeVirt) SetPopulatorPodOwnership(vm *plan.VMStatus) (err error) {
 	return
 }
 
+// DeletePVCs deletes PVCs associated with the VM, including prime PVCs (populator flows)
+func (r *KubeVirt) DeletePVCs(vm *plan.VMStatus) (err error) {
+	pvcs, err := r.getPVCs(vm.Ref)
+	for _, pvc := range pvcs {
+		primePVC := core.PersistentVolumeClaim{}
+		err = r.Destination.Client.Get(context.TODO(), client.ObjectKey{Namespace: r.Plan.Spec.TargetNamespace, Name: fmt.Sprintf("prime-%s", string(pvc.UID))}, &primePVC)
+		if err != nil {
+			continue
```

Comment:
don't we need to ignore IsNotFound here as well?

Reply:
don't we? :)

Reply:
ah I see you've added a check below but we can get it also here from the call to `Get`, no?

Reply:
we can. but in any case of error we will execute `continue` to the loop. the only things that i missed is setting the error to nil, as if it's the last pvc in the loop we will return an error.

Reply:
the problem with this approach is that when the error is not of type IsNotFound, and there's no error later on, we'll return err = nil, how about doing it as we do in quite a few places elsewhere:
```golang
if err != nil {
	if k8serr.IsNotFound(err) {
		err = nil
	}
	continue
}
```

Reply:
I was wrong here, we shouldn't continue when k8serr.IsNotFound(err) != true, posted a fix - @liranr23 please check it out

Reply:
i think we should continue and not return here. we pass on the PVCs if one doesn't have a prime PVC for some reason - fine. let's check the next PVC. why would we just return and skip the rest?

Reply:
we can also take this approach of trying to remove the other items rather then returning and waiting for the next iteration but then the code would become more complex because we'd need to know that there's been an issue in previous iteration, otherwise let's say we failed to get the first prime pvc for some reason and everything afterwards went smoothly - we will return err == nil which doesn't align with the other places in which you return an error

Reply:
> i think we should continue and not return here. we pass on the PVCs if one doesn't have a prime PVC for some reason - fine. let's check the next PVC. why would we just return and skip the rest?

reading this comment again, I see you wrote `if one doesn't have a prime PVC for some reason - fine.` - I agree, the modified code would still do that. at first look I thought you meant to continue when err != nil and !k8serr.IsNotFound(err) == true

Reply:
i think both approaches are overall ok (hitting an error - failing) but i prefer something more into best effort, like: making logs of each error - if we see any. but resuming the deletion? in case we didn't succeed to get the prime we should also avoid trying deleting it?

Reply:
I'm not sure what you mean by "resuming the deletion" but yeah, the modified code now includes an optimization that skips the call to delete when we cannot get the prime pvc

Reply:
and in case we have the PVC (not prime) and  failing to get/delete the prime PVC, why to skip its removal?

Reply:
I'm still not sure I understand what you mean, before the changes I've made we had:
```golang
		// Best effort deletion
		err = r.DeleteObject(&primePVC, vm, "Deleted prime PVC.", "pvc")
		if err != nil && k8serr.IsNotFound(err) {
			err = nil
		}
		if err != nil {
			return
		}
```
do you mean that the last `return` should have been dropped?

wouldn't we have another iteration if we return an error in this function?

Reply:
yeah, in case we fail to delete the prime it might worth to continue and try to delete the PVC itself.

Overall something like this:
In case we didn't get the prime - skip it deletion - try to delete the PVC
In case we didn't succeed to delete the prime - try to delete the PVC.
In case we didn't succeed to delete the PVC - instead of returning, continue to the next PVC.
Log each error in any cases of the above.

Reply:
ok, I looked at the code - in two places we call this function assuming everything there is a best-effort and in one place we check the error, right?
so for the first two, what you write makes sense
but how would it work for the third case?

Reply:
correct. now when i think about it - yes we can say it's best effort for the first two and not for the third.
i don't like the idea of not doing the best effort as it depend where we cancel/fail in the migration to have or not have the prime pvc.

Reply:
@liranr23 I started to change the code to achieve this kind of 'best-effort' attempt to remove entities (it's not just about ignoring the error but also to try to remove as much as possible), see [this commit](https://github.com/kubev2v/forklift/pull/590/commits/9571e3ded1389a9263a1b1035fbf17bfdbcf0a70) - it looks like too much trouble to add logs everywhere and propagate the failOnErr to all places - I'm about to remove it from this PR but if you think it's important, you can do it separately

Reply:
dropped the aforementioned commit, will post it separately

Reply:
gave it another try, @liranr23 please take a look at the last commit